azurerm/internal/services/synapse/synapse_sql_pool_resource.go

```diff
@@ -0,0 +1,378 @@
+		if !utils.ResponseWasNotFound(existing.Response) {
+			return fmt.Errorf("checking for present of existing Synapse SqlPool %q (Resource Group %q / workspaceName %q): %+v", name, workspaceId.ResourceGroup, workspaceId.Name, err)
+		}
+	}
+	if existing.ID != nil && *existing.ID != "" {
+		return tf.ImportAsExistsError("azurerm_synapse_sql_pool", *existing.ID)
+	}
+
+	workspace, err := workspaceClient.Get(ctx, workspaceId.ResourceGroup, workspaceId.Name)
+	if err != nil {
+		return fmt.Errorf("reading Synapse workspace %q (Resource Group %q): %+v", workspaceId.Name, workspaceId.ResourceGroup, err)
+	}
+
+	mode := d.Get("create_mode").(string)
+	sqlPoolInfo := synapse.SQLPool{
+		Location: workspace.Location,
+		SQLPoolResourceProperties: &synapse.SQLPoolResourceProperties{
+			CreateMode: utils.String(mode),
+		},
+		Sku: &synapse.Sku{
+			Name: utils.String(d.Get("sku_name").(string)),
+		},
+		Tags: tags.Expand(d.Get("tags").(map[string]interface{})),
+	}
+
+	switch mode {
+	case "Default":
+		sqlPoolInfo.SQLPoolResourceProperties.Collation = utils.String(d.Get("collation").(string))
+	case "Recovery":
+		sourceDatabaseId := constructSourceDatabaseId(d.Get("source_database_id").(string))
+		if sourceDatabaseId == "" {
+			return fmt.Errorf("`source_database_id` must be set when `create_mode` is `Recovery`")
+		}
+		sqlPoolInfo.SQLPoolResourceProperties.RecoverableDatabaseID = utils.String(sourceDatabaseId)
+	case "PointInTimeRestore":
+		sourceDatabaseId := constructSourceDatabaseId(d.Get("source_database_id").(string))
+		restorePointInTimeStr := d.Get("restore_point_in_time").(string)
+		if sourceDatabaseId == "" || restorePointInTimeStr == "" {
+			return fmt.Errorf("`source_database_id` and `restore_point_in_time` must be set when `create_mode` is `PointInTimeRestore`")
+		}
+		restorePointInTime, _ := time.Parse(time.RFC3339, restorePointInTimeStr)
+		sqlPoolInfo.SQLPoolResourceProperties.RestorePointInTime = &date.Time{Time: restorePointInTime}
+		sqlPoolInfo.SQLPoolResourceProperties.SourceDatabaseID = utils.String(sourceDatabaseId)
+	}
+
+	future, err := sqlClient.Create(ctx, workspaceId.ResourceGroup, workspaceId.Name, name, sqlPoolInfo)
+	if err != nil {
+		return fmt.Errorf("creating Synapse SqlPool %q (Resource Group %q / workspaceName %q): %+v", name, workspaceId.ResourceGroup, workspaceId.Name, err)
+	}
+	if err = future.WaitForCompletionRef(ctx, sqlClient.Client); err != nil {
+		return fmt.Errorf("waiting on creating future for Synapse SqlPool %q (Resource Group %q / workspaceName %q): %+v", name, workspaceId.ResourceGroup, workspaceId.Name, err)
+	}
+
+	if d.Get("data_encrypted").(bool) {
+		parameter := synapse.TransparentDataEncryption{
+			TransparentDataEncryptionProperties: &synapse.TransparentDataEncryptionProperties{
+				Status: synapse.TransparentDataEncryptionStatusEnabled,
+			},
+		}
+		if _, err := sqlPTDEClient.CreateOrUpdate(ctx, workspaceId.ResourceGroup, workspaceId.Name, name, parameter); err != nil {
```

Comment:
since this is a separate client - this'll likely make sense as a separate resource (since it's likely this resource will support MSI / defining a user-defined key to back this in time?)

Reply:
for now, this client could only update `data_encrypted` field, and there is no delete function for it. It's more likely a property of sql pool. So I think  it's better to combine it into this resource ?